sdk/python/tests/compiler/compiler_tests.py

```diff
@@ -334,6 +334,10 @@ def test_py_recursive_while(self):
     """Test pipeline recursive."""
     self._test_py_compile_yaml('recursive_while')
 
+  def test_py_compile_ttl_after_finished(self):
```

Comment:
Can you make the test more focused on the actual introduced behavior?
See `test_set_display_name` for example.

Reply:
- Removed the unnecessary compile test.
- Added an unit test instead.